src/test/ui/span/issue-24356.stderr

```diff
@@ -0,0 +1,10 @@
+error[E0046]: not all trait items implemented, missing: `Target`
+  --> $DIR/issue-24356.rs:30:9
+   |
+30 |         impl Deref for Thing {
+   |         ^ missing `Target` in implementation
+   |
+   = note: `Target` from trait: `type Target;`
```

Comment:
Can you create a test case where the item from the other trait is a `fn` with a non-trivial signature? 

Probably worth testing the constant case too.